bundle/jsky.app.ot/src/main/scala/jsky/app/ot/editor/template/TemplateParametersEditor.scala

```diff
@@ -257,20 +256,25 @@ class TemplateParametersEditor(shells: java.util.List[ISPTemplateParameters]) ex
 
       val JNoneLong: JOption[java.lang.Long] = JNone.instance[java.lang.Long]
 
+      // These are defined in object SPTarget, but private.  Can we open them
+      // up and/or provide get/set for RightAscension/Declination on SPTarget?
+      val ra:  Target @?> RightAscension = Target.coords >=> Coordinates.ra.partial
+      val dec: Target @?> Declination    = Target.coords >=> Coordinates.dec.partial
+
       val hms = new HMSFormat()
-      val raField = new BoundTextField[Double](10)(
-        read = s => hms.parse(s),
-        show = hms.format,
-        get  = _.getTarget.getRaHours(JNoneLong).getOrElse(0.0),
-        set  = setTarget((a, b) => a.setRaHours(b))
+      val raField = new BoundTextField[RightAscension](10)(
+        read = s  => Angle.parseHMS(s).map(RightAscension.fromAngle).valueOr(nfe => throw nfe),
+        show = ra => Angle.formatHMS(ra.toAngle),
+        get  = _.getTarget.getCoordinates(None).map(_.ra).getOrElse(RightAscension.zero),
+        set  = setTarget((a, b) => ra.set(a.getTarget, b).foreach(a.setTarget))
       )
 
       val dms = new DMSFormat()
-      val decField = new BoundTextField[Double](10)(
-        read = s => dms.parse(s),
-        show = dms.format,
-        get  = _.getTarget.getDecDegrees(JNoneLong).getOrElse(0.0),
-        set  = setTarget((a, b) => a.setDecDegrees(b))
+      val decField = new BoundTextField[Declination](10)(
+        read = s   => Angle.parseDMS(s).flatMap(a => Declination.fromAngle(a) \/> new IllegalArgumentException(s"%s is not a valid declination")).valueOr(ex => throw ex),
+        show = dec => Declination.formatDMS(dec),
+        get  = _.getTarget.getCoordinates(None).map(_.dec).getOrElse(Declination.zero),
```

Comment:
Actually I think not here because it should match the Phase I target, we're editing the position and the proper motion is provided as well.